File: core/src/org/tetawex/ecf/util/SavableLevelData.java
package org.tetawex.ecf.util;

import org.tetawex.ecf.model.Cell;
import org.tetawex.ecf.model.LevelData;

/**
 * Created by tetawex on 16.08.17.
 */
public class SavableLevelData {
    private SavableCell[][] cellArray;
    private int mana;
    private int maxScore;
    private String name;


    public SavableLevelData(LevelData data) {
        mana = data.getMana();
        maxScore = data.getMaxScore();
        name = data.getName();

        int width = data.getCellArray().length;
        int height = data.getCellArray()[0].length;
        cellArray = new SavableCell[width][height];

        for (int i = 0; i < width; i++) {
            for (int j = 0; j < height; j++) {
                cellArray[i][j] = new SavableCell(data.getCellArray()[i][j]);
            }
        }
    }

    public LevelData createLevelData() {
        LevelData levelData = new LevelData();
        levelData.setLevelNumber(0);
        levelData.setMaxScore(maxScore);
        levelData.setLevelCode("editor");
        levelData.setMana(mana);
        levelData.setName(name);

        int width = cellArray.length;
        int height = cellArray[0].length;
        Cell[][] trueCellArray = new Cell[width][height];

        for (int i = 0; i < width; i++) {
            for (int j = 0; j < height; j++) {
                trueCellArray[i][j] = cellArray[i][j].createCell();
            }
        }

        levelData.setCellArray(trueCellArray);
        return levelData;
    }
}

File: core/src/org/tetawex/ecf/util/SavableLevelData.kt
package org.tetawex.ecf.util

import org.tetawex.ecf.model.Cell
import org.tetawex.ecf.model.LevelData

/**
 * Created by tetawex on 16.08.17.
 */
class SavableLevelData(data: LevelData) {
    private val cellArray: Array<Array<SavableCell?>>
    private val mana: Int
    private val maxScore: Int
    private val name: String?

    init {
        mana = data.mana
        maxScore = data.maxScore
        name = data.name

        val width = data.cellArray.size
        val height = data.cellArray[0].size
        cellArray = Array(width) { arrayOfNulls<SavableCell>(height) }

        for (i in 0 until width) {
            for (j in 0 until height) {
                cellArray[i][j] = SavableCell(data.cellArray[i][j])
            }
        }
    }

    fun createLevelData(): LevelData {
        val levelData = LevelData()
        levelData.levelNumber = 0
        levelData.maxScore = maxScore
        levelData.levelCode = "editor"
        levelData.mana = mana
        levelData.name = name

        val width = cellArray.size
        val height = cellArray[0].size
        val trueCellArray = Array<Array<Cell?>>(width) { arrayOfNulls(height) }

        for (i in 0 until width) {
            for (j in 0 until height) {
                trueCellArray[i][j] = cellArray[i][j]!!.createCell()
            }
        }

        levelData.cellArray = trueCellArray
        return levelData
    }
}
